core/src/main/java/org/jahia/modules/modulemanager/flow/ModuleManagementFlowHandler.java

```diff
@@ -176,16 +178,41 @@ public boolean uploadModule(MultipartFile moduleFile, MessageContext context, bo
                     .code("serverSettings.manageModules.install.moduleFileRequired").build());
             return false;
         }
+
         String originalFilename = moduleFile.getOriginalFilename();
-        if (!FilenameUtils.isExtension(StringUtils.lowerCase(originalFilename), "jar")) {
-            context.addMessage(new MessageBuilder().error().source("moduleFile")
-                    .code("serverSettings.manageModules.install.wrongFormat").build());
-            return false;
-        }
+        boolean canHandleExtension = FilenameUtils.isExtension(StringUtils.lowerCase(originalFilename), "jar");
         File file = null;
         try {
             file = File.createTempFile("module-", "." + StringUtils.substringAfterLast(originalFilename, "."));
             moduleFile.transferTo(file);
+
+            //If it cannot be handled as Jar, look for impl that can handle this extension
+            if(!canHandleExtension) {
+                BundleContext bundleContext = FrameworkService.getBundleContext();
+                Collection<ServiceReference<ArtifactUrlTransformer>> serviceReferences = bundleContext.getServiceReferences(ArtifactUrlTransformer.class, null);
+                for (ServiceReference<ArtifactUrlTransformer> serviceReference : serviceReferences) {
+                    ArtifactUrlTransformer transformer = bundleContext.getService(serviceReference);
+                    if (transformer.canHandle(file)) {
+                        File fileTemp = File.createTempFile("module-", "." + StringUtils.substringAfterLast(originalFilename, "."));
+                        try {
+                            URL transformedURL = transformer.transform(new URL("file:" + file.getPath()));
+                            FileUtils.copyInputStreamToFile(transformedURL.openConnection().getInputStream(), fileTemp);
+                            installBundles(fileTemp, context, originalFilename, forceUpdate, autoStart);
+                            return true;
+                        } finally {
+                            FileUtils.deleteQuietly(fileTemp);
+                        }
+                    }
+                }
+            }
+
+            //If it is not a jar and none impl can handle this extension
+            if (!canHandleExtension) {
+                context.addMessage(new MessageBuilder().error().source("moduleFile")
+                        .code("serverSettings.manageModules.install.wrongFormat").build());
+                return false;
+            }
+
             installBundles(file, context, originalFilename, forceUpdate, autoStart);
             return true;
```

Comment:
We should delete file also? @rfonseca85 @tdraier

Reply:
file is being deleted as well in the finally of uploadModule method

```
        } finally {
            FileUtils.deleteQuietly(file);
        }
```